File: Git-Templates/ProjectLanguageFiles/Education/!Learn/Learn-Mercury/Mercury/PROJECT_LANG_1.m
% Start of script
:- module project-language-file-one.
:- interface.
:- import_module io.
:- pred main(io::di, io::uo) is det.

:- implementation.
main(!IO) :-
	io.write_string("Project language file 1\nFor: seanpm2001/Learn-Mercury\nAbout:\nI chose Mercury as the first project language file for this project (Seanpm2001/Learn-Mercury) as this project is about learning the Mercury programming language, and showing my knowledge for the language. Its project language file should represent what language is being showcased and studied here\n", !IO).

% Output:

% Project language file 1
% For: seanpm2001/Learn-Mercury
% About:
% I chose Mercury as the first project language file for this project (Seanpm2001/Learn-Mercury) as this project is about learning the Mercury programming language, and showing my knowledge for the language. Its project language file should represent what language is being showcased and studied here

% File info
% File type: Mercury source file (*.m)
% File version: 1 (2022, Friday, April 15th at 9:21 pm PST)
% Line count (including blank lines and compiler line): 24

% End of script

File: Git-Templates/ProjectLanguageFiles/Education/!Learn/Learn-R/R/PROJECT_LANG_1.r
#!/usr/bin/env Rscript
# Project language file 1
# For: seanpm2001/Learn-R
# About
# I chose R as the first project language file for this project (Seanpm2001/Learn-R) as this project is about learning the R programming language, and showing my knowledge for the language. Its project language file should represent what language is being showcased and studied here.
# Now time for a Mandelbrot (From Wikipedia/wiki/R/)

install.packages("caTools")  # install external package
library(caTools)             # external package providing write.gif function
jet.colors <- colorRampPalette(c("red", "blue", "#007FFF", "cyan", "#7FFF7F",
                                 "yellow", "#FF7F00", "red", "#7F0000"))
dx <- 1500                    # define width
dy <- 1400                    # define height
C  <- complex(real = rep(seq(-2.2, 1.0, length.out = dx), each = dy),
              imag = rep(seq(-1.2, 1.2, length.out = dy), dx))
C <- matrix(C, dy, dx)       # reshape as square matrix of complex numbers
Z <- 0                       # initialize Z to zero
X <- array(0, c(dy, dx, 20)) # initialize output 3D array
for (k in 1:20) {            # loop with 20 iterations
  Z <- Z^2 + C               # the central difference equation
  X[, , k] <- exp(-abs(Z))   # capture results
}
write.gif(X, "Mandelbrot.gif", col = jet.colors, delay = 100)

# File info
# File type: R Source file (*.r)
# File version: 1 (2022, Wednesday, April 20th at 2:03 pm)
# Line count (including blank lines and compiler line): 30
# End of script
